File: data structure/binarysearch.c
#include <stdio.h>
int binarysearch(int arr[], int size, int element)
{
    int low, mid, high;
    low = 0;
    high = size - 1;
    //keep searching until low<=high
    while (low <= high)
    {
        mid = (low + high) / 2;
        if (arr[mid] == element)
        {
            return mid;
        }
        if (arr[mid] < element)
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
    }
    return -1;
}
int main()
{
    int arr[] = {1, 3, 4, 5, 6, 7, 8, 12, 23, 45, 56};
    int size = sizeof(arr) / sizeof(int);
    int element = 4;
    int searchIndex = binarysearch(arr, size, element);
    printf("The element %d was found at index %d \n",element, searchIndex);
    return 0;
}

File: Python/binarysearch.py
# -*- coding: utf-8 -*-
"""
Created on Tue Dec 14 16:30:24 2021

@author: Souvik
"""

def binary_search(list1, n):  
    low = 0  
    high = len(list1) - 1  
    mid = 0  
  
    while low <= high:  
        
        mid = (high + low) // 2  
  
          
        if list1[mid] < n:  
            low = mid + 1  
  
          
        elif list1[mid] > n:  
            high = mid - 1  
  
        else: 
            return mid;
             
      
              
    return -1  
  
  
list1 = []
list1=[item for item in input("Enter the list items : ").split()]
print(list1)
n = input("Enter item to search : ")  
  

result = binary_search(list1, n)  
  
if result != -1:  
    
    print("Element is present in the given list  at index", str(result)) 
else:  
    print("Element is not present in the given list.")
